### PR TITLE
Rename Trace* config parameters to DriverLog* to fix iODBC tracing vs. driver's own logging

### PR DESCRIPTION
**Pull request by traceon**
(somewhat) BREAKING CHANGE!

Rename driver-specific parameters to avoid clashes with system (iODBC Driver Manager) parameters:
    Trace to Log
    TraceFile to LogFile

Define and use custom [connection] attributes to avoid clashes with system (iODBC Driver Manager) attributes:
    SQL_ATTR_TRACE to CH_SQL_ATTR_TRACE
    SQL_ATTR_TRACEFILE to CH_SQL_ATTR_TRACEFILE

Duplicate [ODBC] section in configs and config samples to cover iODBC behaviour

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -208,16 +212,16 @@ script:
     Description = ClickHouse DSN (ANSI, localhost)
     Url         = http://${CLICKHOUSE_SERVER_IP}
     Timeout     = 30
-    Trace       = yes
-    TraceFile   = /tmp/clickhouse-odbc-driver-trace.log
+    Log         = yes
```
Comment (filimonov) [inline .travis.yml]: Maybe something less general? Like DebugLog?

File: driver/api/impl/attr.cpp [hunk 2]
```diff
@@ -176,24 +176,27 @@ SQLRETURN GetConnectAttr(
...
-                SQL_ATTR_LOGIN_TIMEOUT, SQLUSMALLINT, connection.session ? connection.session->getTimeout().seconds() : connection.timeout);
+            CASE_NUM(SQL_ATTR_LOGIN_TIMEOUT, SQLUSMALLINT, connection.session ? connection.session->getTimeout().seconds() : connection.timeout);
             CASE_NUM(SQL_ATTR_TXN_ISOLATION, SQLINTEGER, SQL_TXN_SERIALIZABLE); // mssql linked server
             CASE_NUM(SQL_ATTR_AUTOCOMMIT, SQLINTEGER, SQL_AUTOCOMMIT_ON);
-            CASE_NUM(SQL_ATTR_TRACE, SQLINTEGER, (connection.getDriver().isLoggingEnabled() ? SQL_OPT_TRACE_ON : SQL_OPT_TRACE_OFF));
 
             case SQL_ATTR_CURRENT_CATALOG:
-                fillOutputPlatformString(connection.getDatabase(), out_value, out_value_max_length, out_value_length);
-                return SQL_SUCCESS;
+                return fillOutputPlatformString(connection.getDatabase(), out_value, out_value_max_length, out_value_length);
```
Comment (filimonov) [inline driver/api/impl/attr.cpp]: Is it possible that result was ignored and reported as success every time for some reason?

File: .travis.yml (hunk 1 above)
Comment (traceon) [reply .travis.yml]: Technically, it enables/redirects any driver-specific log. Verbosity, when and if implemented, may only control it further.

File: driver/api/impl/attr.cpp (hunk 2 above)
Comment (traceon) [reply driver/api/impl/attr.cpp]: Yes...

File: driver/config/ini_defines.h [hunk 3]
```diff
@@ -45,9 +45,9 @@
 #endif
 
 #ifdef _win_
-#    define INI_TRACEFILE_DEFAULT "\\temp\\clickhouse-odbc.log"
+#    define INI_TRACEFILE_DEFAULT "\\temp\\clickhouse-odbc-driver.log"
```
Comment (filimonov) [inline driver/config/ini_defines.h]: Not sure if it's a valid path for Windows. But not sure if  we can use env here, with smth like "%TEMP%\clickhouse-odbc.log"

File: .travis.yml [hunk 4]
```diff
@@ -199,6 +199,10 @@ script:
     EOF
   - |
     cat > $ODBCSYSINI/.odbc.ini <<-EOF
+    [ODBC]
+    Trace     = yes
+    TraceFile = /tmp/odbc-driver-manager.log
```
Comment (filimonov) [inline .travis.yml]: since we create those file on travis, can we can that they appeared after our tests?

File: driver/config/ini_defines.h (hunk 3 above)
Comment (traceon) [reply driver/config/ini_defines.h]: I believe, originally it was inspired by this:
https://github.com/lurcher/unixODBC/blob/4209bc229c64a0e00920dac635e666d2233c794f/include/sqlext.h#L194

File: .travis.yml (hunk 4 above)
Comment (traceon) [reply .travis.yml]: Yes, I can incorporate that in the upcoming travis changes:
https://github.com/ClickHouse/clickhouse-odbc/pull/198

File: driver/config/ini_defines.h (hunk 3 above)
Comment (filimonov) [reply driver/config/ini_defines.h]: OK. We can check if it works. Someday :) Not important for now.

File: .travis.yml (hunk 1 above)
Comment (traceon) [reply .travis.yml]: Renamed to `DriverLog` and `DriverLogFile`.

**Pull request merged**